### PR TITLE
[Spark] Rename managed commit to coordinated commits

### PR DESCRIPTION
**Pull request by dhruvarya-db**
#### Which Delta project/connector is this regarding?

- [x] Spark
- [ ] Standalone
- [ ] Flink
- [ ] Kernel
- [x] Other (RFC)

## Description

Renames Managed Commit to Coordinated Commits and Commit Owner to Commit Coordinator to better express the meaning of the feature. Configs, table feature name, classes, functions, comments, file names, and directory names have all been updated to reflect the new terminology.

## How was this patch tested?

Existing tests should cover these changes.

## Does this PR introduce _any_ user-facing changes?

Yes, the feature name and config names have changed.

### CODE REVIEW

**Review by prakharjain09 — APPROVED** [review]

File: connectors/standalone/src/main/scala/io/delta/standalone/internal/OptimisticTransactionImpl.scala [hunk 1]
```diff
@@ -527,7 +527,9 @@ private[internal] class OptimisticTransactionImpl(
   private def shouldCheckpoint(committedVersion: Long): Boolean = {
     val checkpointingEnabled =
       deltaLog.hadoopConf.getBoolean(StandaloneHadoopConf.CHECKPOINTING_ENABLED, true)
-    checkpointingEnabled && committedVersion != 0 && committedVersion % deltaLog.checkpointInterval == 0
+    checkpointingEnabled &&
```
Comment (scottsand-db) [inline connectors/standalone/src/main/scala/io/delta/standalone/internal/OptimisticTransactionImpl.scala:530]: Why this change?

Comment (dhruvarya-db) [reply connectors/standalone/src/main/scala/io/delta/standalone/internal/OptimisticTransactionImpl.scala]: My bad, I have reverted this change.
